### PR TITLE
Fix for invalid byte sequence in UTF-8 for command_output

### PR DESCRIPTION
**Pull request by ark-konopacki**
replaced due to ruby doc:
```
Note that while forceencoding simply tags the String with a different encoding, encode converts the bytes of one encoding into the equivalent bytes of the second. As a result, while forceencoding should almost never be used unless you know for sure that the bytes actually represent the characters you want in the target encoding, encode is relatively safe to use to convert a String into the encoding you want.
```
I think we should replace all force_encode occurrence in run_loop 

i hope it fix https://github.com/calabash/run_loop/issues/373

### CODE REVIEW

**Comment by jmoody** [comment]
Jenkins test this please

File: lib/run_loop/xcrun.rb [hunk 1]
```diff
@@ -59,7 +59,7 @@ def exec(args, options={})
         command_output = CommandRunner.run(['xcrun'] + args, timeout: timeout)
 
         if command_output[:out]
-          out = command_output[:out].force_encoding('utf-8').chomp
+          out = command_output[:out].encode('UTF-8', 'UTF-8', :invalid => :replace).chomp
```
Comment (jmoody) [inline lib/run_loop/xcrun.rb]: Gotta have some rspec tests for this somehow.

I would mock the CommandRunner output.

```
let (:process_status) do
   Class.new do
      def to_s; '#<Mock ProcessStatus>'; end
      def inspect; to_s; end
      def exitstatus; 0; end
    end.new
end

it "handles UTF-8 encoding" do
args = ["xcrun", "sample"]
options = { :timeout => 2.0 }
hash = {
   :out => "some string with some non-UTF8 character that will be replaced",
   :pid => 11,
   :status => process_status.new
}
expect(CommandRunner).to receive(:run).with(args, options).and_return(hash)

actual = Xcrun.new.exec("sample")

expect(actual[:out]).to be == "string with unknown UTF-8 character replaced with `''`"
end
```

**Comment by ark-konopacki** [comment]
@jmoody I update implementation it can be slow, but I tests it with different bad UTF-8 strings  and it works as expected. I cannot create additional test(s) for that implementation i'm not so familiar with mocking in tests.

**Comment by jmoody** [comment]
This is not perfect, but it is a step in the right direction.

For various reasons, I cannot update this PR so I will merge and make a new PR with fixes.

**Comment by jmoody** [comment]
The Travis failure is because rvm can't install ruby 2.1.7

**Pull request merged**